### PR TITLE
Added new applications

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Added sssp from UW documentation

### CODE REVIEW

File: apps/python_bindings/sssp/sssp.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import graphit
```
Comment (yunmingzhang17) [inline apps/python_bindings/sssp/sssp.py]: Could you add an acknowledgement to the UW folks for these files in the comments? Take a look at their fork of GraphIt to figure out who wrote what parts?

File: apps/python_bindings/sssp/sssp_export.gt [hunk 2]
```diff
@@ -0,0 +1,46 @@
+element Vertex end
```
Comment (yunmingzhang17) [inline apps/python_bindings/sssp/sssp_export.gt]: Add some ack in the comments for this file as well?

File: src/python/graphit.py [hunk 3]
```diff
@@ -84,6 +84,101 @@ def compile_and_load(graphit_source_file, extern_cpp_files=[], linker_args=[], p
...
+	module_filename_so = module_filename_base + ".so"
+
+	# check if it is loaded before and it is the latest, we return the cached version.
+	if os.path.exists(module_filename_so) and os.stat(graphit_source_file).st_mtime < os.stat(module_filename_so).st_mtime:
+		spec = importlib.util.spec_from_file_location(module_name, module_filename_so)
+		module = importlib.util.module_from_spec(spec)
+		spec.loader.exec_module(module)
+		return module
+	
+	#else we just follow the normal routine
```
Comment (yunmingzhang17) [inline src/python/graphit.py]: don't duplicate the compile_and_load code, just invoke the compile_and_load code. Duplication leads to potential inconsistencies in the future.

File: src/python/graphit.py [hunk 4]
```diff
@@ -84,6 +84,101 @@ def compile_and_load(graphit_source_file, extern_cpp_files=[], linker_args=[], p
...
+	try:
+		subprocess.check_call("python " + GRAPHIT_BUILD_DIRECTORY + "/bin/graphitc.py -f " + graphit_source_file + " -o " + module_filename_cpp + " -m " + module_name, stderr=subprocess.STDOUT, shell=True)
```
Comment (yunmingzhang17) [inline src/python/graphit.py]: Can you add some unit tests for the compile_and_load_cache method in pybind_test.py (https://github.com/GraphIt-DSL/graphit/blob/master/test/python/pybind_test.py)? And try to add some unit tests for the two files in the apps/python_binding directory in pybind_test.py?

File: apps/python_bindings/pagerank_delta/pagerank_delta_export.gt [hunk 5]
```diff
@@ -1,3 +1,6 @@
+% This file is copied from a repo - (https://github.com/bespoke-silicon-group/hb_starlite/tree/master/py-graphit-example)
```
Comment (yunmingzhang17) [inline apps/python_bindings/pagerank_delta/pagerank_delta_export.gt]: This file is not from the UW folks right (originally developed by us)? There shouldn't be an ack here?

File: test/input/export_pagerank_delta_UW.gt [hunk 6]
```diff
@@ -0,0 +1,72 @@
+% This file is copied from a repo - (https://github.com/bespoke-silicon-group/hb_starlite/tree/master/py-graphit-example)
```
Comment (yunmingzhang17) [inline test/input/export_pagerank_delta_UW.gt]: Similar to the above comment, I think this file is from our own pagerankdelta implementation? Or is this also from UW folks?

File: test/python/pybind_test.py [hunk 7]
```diff
@@ -140,6 +165,8 @@ def test_pybind_constant_size_vector_of_vector_return(self):
         self.assertEqual(vector_return.shape, (10, 10))
         self.assertEqual(np.sum(vector_return), 550)
 
+    #TODO test examples in python_bindings directory (currently includes pagerank_delta, sssp)
```
Comment (yunmingzhang17) [inline test/python/pybind_test.py]: Are we going to fix this TODO?

File: test/input/export_pagerank_delta_UW.gt (hunk 6 above)
Comment (tugsbayasgalan) [reply test/input/export_pagerank_delta_UW.gt]: this is from UW folks

File: apps/python_bindings/pagerank_delta/pagerank_delta_export.gt (hunk 5 above)
Comment (tugsbayasgalan) [reply apps/python_bindings/pagerank_delta/pagerank_delta_export.gt]: This is also from UW folks. I think their implementation was slightly different from ours

File: test/python/pybind_test.py (hunk 7 above)
Comment (tugsbayasgalan) [reply test/python/pybind_test.py]: Ah this should be gone! i added the tests above. I was thinking about testing their python binding code but i figured it makes more sense to test the graphit code.

File: test/input/export_pagerank_delta_UW.gt (hunk 6 above)
Comment (yunmingzhang17) [reply test/input/export_pagerank_delta_UW.gt]: But I don't see the pagerankdelta program in the link in the comment? There's only sssp?

File: apps/python_bindings/pagerank_delta/pagerank_delta_export.gt (hunk 5 above)
Comment (tugsbayasgalan) [reply apps/python_bindings/pagerank_delta/pagerank_delta_export.gt]: Ah my bad! this is indeed ours. Should be fixed now

**Pull request merged**